wasmer/instance.go

```diff
@@ -167,11 +167,23 @@ func newInstanceWithImports(
 		return emptyInstance, err
 	}
 
-	var exports = make(map[string]func(...interface{}) (Value, error))
+	exports, hasMemory, err := getExportsFromInstance(instance, &memory)
+	if err != nil {
+		return emptyInstance, err
+	}
+	
+	if hasMemory == false {
+		return Instance{instance: instance, imports: imports, Exports: exports, Memory: nil}, nil
+	}
 
-	var wasmExports *cWasmerExportsT
-	var hasMemory = false
+	return Instance{instance: instance, imports: imports, Exports: exports, Memory: &memory}, nil
+}
 
+// Returns the exports, whether it has memory or an error
+func getExportsFromInstance(instance *cWasmerInstanceT, memory *Memory) (map[string]func(...interface{}) (Value, error), bool, error) {
+	var exports = make(map[string]func(...interface{}) (Value, error))
+	var hasMemory = false
```

Comment:
I think `hasMemory` might be better calculated in a function apart?

Reply:
Or rather than in a function apart, just checking if in the returned `Instance` , the memory is null?

Reply:
Yeah, that's reasonable -- I just kept the existing structure when I copied and pasted that inner part out; checking for null should work though!